benchmarks/ALCE/ELI5/utils.py

```diff
@@ -0,0 +1,60 @@
+import json
+
+
+def make_doc_prompt(doc, doc_id, doc_prompt, method=None):
+    # For doc prompt:
+    # - {ID}: doc id (starting from 1)
+    # - {T}: title
+    # - {P}: text
+    # use_shorter: None, "summary", or "extraction"
+
+    if method == "vanilla":
+        text = doc['text']
+    elif method == "summary":
+        text = doc["summary"]
+    elif method == "snippet":
+        text = doc["extraction"]
+    else:
+        raise ValueError("Don't support such method.")
+    return doc_prompt.replace("{T}", doc["title"]).replace("{P}", text).replace("{ID}", str(doc_id+1))
+
+
+def make_demo(item, prompt, ndoc=None, doc_prompt=None, instruction=None, method=None, test=False):
+    # For demo prompt
+    # - {INST}: the instruction
+    # - {D}: the documents
+    # - {Q}: the question
+    # - {A}: the answers
+    # ndoc: number of documents to put in context
+    # use_shorter: None, "summary", or "extraction"
+
+    prompt = prompt.replace("{INST}", instruction).replace("{Q}", item['question'])
+    doc_texts = []
+    if "{D}" in prompt:
+        if ndoc == 0:
```

Comment:
If `ndoc` is an `Optional[int]` type, type checking should be performed before. Otherwise, it should be an `int` type and should not default to `None`.

Reply:
> If `ndoc` is an `Optional[int]` type, type checking should be performed before. Otherwise, it should be an `int` type and should not default to `None`.

In fact, `ndoc` has been set to `int` type in the `run.py` file, and the default value is set to 5. There is no need to worry about `ndoc` being empty or having an incorrect type.

Reply:
Of course, this issue should really be considered if other files use this function again.

Reply:
> > If `ndoc` is an `Optional[int]` type, type checking should be performed before. Otherwise, it should be an `int` type and should not default to `None`.
> 
> In fact, `ndoc` has been set to `int` type in the `run.py` file, and the default value is set to 5. There is no need to worry about `ndoc` being empty or having an incorrect type.

I knew this, but to ensure readability, it's better not to assign `None` to `ndoc` in the function parameters.

Reply:
> > > If `ndoc` is an `Optional[int]` type, type checking should be performed before. Otherwise, it should be an `int` type and should not default to `None`.
> > 
> > 
> > In fact, `ndoc` has been set to `int` type in the `run.py` file, and the default value is set to 5. There is no need to worry about `ndoc` being empty or having an incorrect type.
> 
> I knew this, but to ensure readability, it's better not to assign `None` to `ndoc` in the function parameters.

Yes, I think it's sound advice.